samza-core/src/main/scala/org/apache/samza/job/local/ProcessJobFactory.scala

```diff
@@ -52,54 +53,62 @@ class ProcessJobFactory extends StreamJobFactory with Logging {
 
     val configFromCoordinatorStream: Config = CoordinatorStreamUtil.readConfigFromCoordinatorStream(coordinatorStreamStore)
 
-    val changelogStreamManager = new ChangelogStreamManager(new NamespaceAwareCoordinatorStreamStore(coordinatorStreamStore, SetChangelogMapping.TYPE))
+    val systemAdmins = new SystemAdmins(configFromCoordinatorStream)
 
-    val coordinator = JobModelManager(configFromCoordinatorStream, changelogStreamManager.readPartitionMapping(), coordinatorStreamStore, metricsRegistry)
-    val jobModel = coordinator.jobModel
+    try {
+      systemAdmins.start()
```

Comment:
Should this move out of the try-catch? It looks like the pattern you are using for lifecycle management of `systemAdmins` is a bit inconsistent across classes.
In some classes, you put the `start` inside the try-finally which has the `stop`, but in other classes you have the `start` outside. Could you please try to make it consistent?

Reply:
Done.